scalyr_agent/agent_main.py

```diff
@@ -239,7 +241,7 @@ def main(self, config_file_path, command, command_options):
         quiet = command_options.quiet
         verbose = command_options.verbose
         status_format = command_options.status_format
-        no_fork = command_options.no_fork
+        self.__no_fork = command_options.no_fork
```

Comment:
It's a minor point, but I think it's better to not introduce a state variable like this when it's not really needed.  I think, you just need to add a `no_fork` argument to the `_perform_config_checks` method and then you don't need to create a `self.__no_fork` variable.  I believe it also results in less lines needing to be changed.  Let me know if that's not correct.

Reply:
It would also need to be passed in to `__run`, and that method is used as a parameter to `__controller.start_agent_service`, we have fork info in there as well but that does add more places to pass parameters, any `platform_controller` implementations. I'm not sure if its better to avoid passing the info around in these places or avoid the extra state. What do you think?

Reply:
Ok, never mind then.  I would say though is then remove `no_fork` as the parameter for the other methods like `__start`, `__restart`, etc.  We should at least be consistent -- either we are passing it around or relying on it to be there.